### PR TITLE
[Breaking changes] Rework combobox-nav APIs and use ARIA 1.2

### PR DESCRIPTION
**Pull request by muan**
Summary of changes:

- `install()` has been renamed to `start()`, for use when options shows up
- `uninstall()` has been renamed `stop()`, for use when options are removed
- Newly added `install()` sets required ARIA attributes, for initialization
- Newly added `uninstall()` removes ARIA attributes, for breakdown

@jscholes This now adds an additional (required) call to set ARIA attributes; this was previously done inside of `<auto-complete>`'s element connected callback. I have one remaining question: Spec says to set `aria-haspopup`, but we've previously [gotten the feedback](https://github.com/github/details-dialog-element/issues/34) that [support for the attribute is poor for non-menu values](https://a11ysupport.io/tech/aria/aria-haspopup_attribute#expectations). Would you recommend that we leave that out still here? **EDIT**: looks like the spec says it's already implicitly set to listbox for elements with combobox role.

### CODE REVIEW

**Comment by jscholes** [comment]
@muan This looks great!  Happy to test when you'd like me to. 
aria-haspopup is definitely problematic in a number of cases, mostly 
because when you use values such as "dialog" on roles like buttons, many 
screen readers will still treat it as "menu".  But for combo boxes, the 
screen reader should do the right thing, because specifying that a combo 
box has a sub-menu doesn't really make any sense in the first place.  So 
the same problem doesn't occur.

TL;DR: use aria-haspopup on combo boxes :)

**Comment by muan** [comment]
@jscholes Thanks! There are some more structural changes we want to make so I'm going to merge this first without cutting a release so avoid a giant PR. I'll ping you once those are reviewed, and definitely before we publish a new release.
